source/server/server.cc

```diff
@@ -191,6 +191,14 @@ void InstanceImpl::initialize(Options& options, TestHooks& hooks,
   admin_.reset(new AdminImpl(initial_config.admin().accessLogPath(),
                              initial_config.admin().profilePath(), initial_config.admin().address(),
                              *this));
+
+  if (options.adminAddressPath().length() > 0) {
+    std::ofstream admin_address_file(options.adminAddressPath());
+    ASSERT(admin_address_file.is_open());
```

Comment:
There should be better error handling here - it's easy for the user to supply a bad path, somewhere not writable, etc. We should `log().criticial` the failure and return false (add return code). The exception catching in the caller is too focused on the JSON case, there are other reasons we can fail here, so that should also be refactored to reflect that.